src/lib/components/select/QSelect.svelte

```diff
@@ -153,47 +104,86 @@
   });
 </script>
 
-<div bind:this={wrapper} class={classes} {...$$restProps}>
-  <slot name="prepend" />
-
-  <input
-    type="text"
-    on:focus={(e) => handleFocus(e.currentTarget)}
-    on:blur={(e) => handleBlur(e.currentTarget)}
-    on:mousedown={() => handleMousedown()}
-    bind:value
-    bind:this={inputElement}
-    tabindex={disable === true ? -1 : 0}
-    readonly
-    disabled={disable}
-  />
-
-  <QIcon
-    class="q-select__arrow-toggle {$$slots.append ? 'q-select__arrow-toggle--has-append' : ''}"
-    name={`arrow_drop_${isMenuOpen ? "up" : "down"}`}
-  />
-
-  <slot name="append" />
-
-  <div class="q-select__menu {isMenuOpen ? 'q-select__menu--active' : ''}">
-    {#each options as option, idx}
-      <a
-        href={multiple ? "javascript:void(0)" : undefined}
-        class="q-select__option {selectedOptions[idx] ? 'q-select__option--selected' : ''}"
-        on:mousedown={() => (preventClose = true)}
-        on:click={(e) => select(e, option)}>{typeof option === "string" ? option : option.value}</a
-      >
-    {/each}
-  </div>
-
-  {#if label}
-    <!-- svelte-ignore a11y-label-has-associated-control -->
-    <label class="q-select__label {active ? 'q-select__label--active' : ''}">{label}</label>
+<div
+  class="q-select q-field {userClasses}"
+  class:q-field--default={!outlined && !rounded && !filled}
+  class:q-field--outlined={outlined}
+  class:q-field--rounded={rounded}
+  class:q-field--filled={filled}
+  class:q-field--has-border={outlined || rounded}
+  class:q-field--dense={dense}
+  class:q-field--active={active}
+  class:q-field--focus={focus}
+  class:q-field--label={label}
+  class:q-field--slot-append={$$slots.append}
+  class:q-field--slot-prepend={$$slots.prepend}
+  class:q-field--disable={disable}
+  class:q-field--error={error}
+  style="--slot-prepend-width: {slotPrependWidth}px"
+  {...$$restProps}
+>
+  {#if $$slots.before}
+    <div class="q-field__slot-before">
+      <slot name="before" />
+    </div>
   {/if}
 
-  {#if hint}
-    <span class="q-select__helper">{hint}</span>
-  {:else if error && errorMessage}
-    <span class="q-select__error">{errorMessage}</span>
+  <div class="q-field__inner">
+    <label class="q-field__wrapper">
+      {#if $$slots.prepend}
+        <div class="q-field__slot-prepend" bind:clientWidth={slotPrependWidth}>
+          <slot name="prepend" />
+        </div>
+      {/if}
+
+      <input
+        class="q-field__input"
+        bind:value
+        placeholder=""
+        on:focus={(e) => handleFocus(e.currentTarget)}
```

Comment:
Prefer a function that takes the event as the parameter and take the target inside it. You could do it like this:
```js
function handleFocus({ currentTarget }: FocusEvent) { // Or whatever type that is
  // ...
}
```
So you can do `on:focus={handleFocus}`